KOALA-3926: Questionnaire and QuestionnaireResponse changes for USCDI v3 requirements

milosdjakovic-vicert opened the pull request:
Linked Issue
------------

https://canvasmedical.atlassian.net/browse/KOALA-3926

Linked PRs
------------

- canvas: https://github.com/canvas-medical/canvas/pull/18545
- fumage: https://github.com/canvas-medical/fumage/pull/879

Description
-----------


[KOALA-3926]: https://canvasmedical.atlassian.net/browse/KOALA-3926?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

milosdjakovic-vicert committed:
docs(questionnaireresponse): add identifier and valueDecimal fields

Add documentation for new US Core fields. Both are READ-only
(exclude_in: create, update).

flackjapz commented on collections/_api/questionnaireresponse.md:
```diff
@@ -27,6 +27,33 @@ sections:
+                type: string
+                description: Unique id for inter-element referencing.
+              - name: use
+                type: enum [ usual | official | temp | secondary | old ]
```
Use `enum_options` instead of this.

Look for other resources as examples that are using that.

flackjapz commented on collections/_api/questionnaireresponse.md:
```diff
@@ -27,6 +27,33 @@ sections:
+                description: The purpose of this identifier.
+              - name: system
+                type: string
+                description: The namespace for the identifier value.
```
Add example system that Canvas will be / could be returning, of course, if you/we know which one we should be using. I just left an accompanying comment in Canvas PR for this, in the build ONC script.

milosdjakovic-vicert committed:
docs(questionnaireresponse): improve identifier field documentation

- Convert `use` field from inline enum to `enum_options` format
- Add `enum_options` with example system URI for `system` field

csande committed:
Merge branch 'main' into feature/koala-3926-questionnaire-response-new-fields

csande committed:
Merge branch 'main' into feature/koala-3926-questionnaire-response-new-fields

csande committed:
Reverted changes

csande committed:
Nested items for FHIR Questionnaire; nested items, identifier, valueDecimal answers, questionnaire name extension, and non-FHIR questionnaire URL extension for FHIR QuestionnaireResponse

csande committed:
Update collections/_api/questionnaireresponse.md

Co-authored-by: Andrew Duane <andrew.duane@canvasmedical.com>

csande committed:
Merge branch 'main' into feature/koala-3926-questionnaire-response-new-fields

rmagier1 commented on collections/_api/questionnaireresponse.md:
```diff
@@ -171,6 +211,14 @@ sections:
+                - name: item
+                  type: array[json]
+                  description: >-
+                    Nested questionnaire response items. This `item` attribute is nested underneath another `item` attribute, meaning that the containing `item` represents a group.
```
Again just be specific that these child attributes match the patient `item` attributes

csande committed:
Note that responses to external questionnaires will not result in a command inserted into a note.

csande committed:
Added comment clarifying that the schema for nested items is the same as for items at the root

csande committed:
Exclude identifier for update

csande committed:
Break out elements of questionnaire extension description

csande committed:
For QuestionnaireResponse, added text to nested item description clarifying that the schema for nested items is the same as for items at the root

csande committed:
Merge branch 'main' into feature/koala-3926-questionnaire-response-new-fields

csande committed:
Merge branch 'main' into feature/koala-3926-questionnaire-response-new-fields

csande committed:
Merge branch 'main' into feature/koala-3926-questionnaire-response-new-fields
